src/ng/compile.js

```diff
@@ -2170,6 +2172,20 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
     }
 
 
+    function downloadTemplateDependencies(directive) {
```

Comment:
you mean it would end up in an infinite loop, right? you are right, but let's discuss a bit if we want this feature before I put more effort into this.